testing/python/collect.py

```diff
@@ -1094,37 +1116,37 @@ def pytest_pycollect_makeitem(collector, name, obj):
                     return MyFunction.from_parent(name=name, parent=collector)
         """
         )
-        item = testdir.getitem("def test_func(): pass")
+        item = pytester.getitem("def test_func(): pass")
         item.config.pluginmanager.getplugin("runner")
         assert item.location == ("ABCDE", 42, "custom")
 
-    def test_func_reportinfo(self, testdir):
-        item = testdir.getitem("def test_func(): pass")
+    def test_func_reportinfo(self, pytester: Pytester) -> None:
+        item = pytester.getitem("def test_func(): pass")
         fspath, lineno, modpath = item.reportinfo()
         assert fspath == item.fspath
         assert lineno == 0
         assert modpath == "test_func"
 
-    def test_class_reportinfo(self, testdir):
-        modcol = testdir.getmodulecol(
+    def test_class_reportinfo(self, pytester: Pytester) -> None:
+        modcol = pytester.getmodulecol(
             """
             # lineno 0
             class TestClass(object):
                 def test_hello(self): pass
         """
         )
-        classcol = testdir.collect_by_name(modcol, "TestClass")
-        fspath, lineno, msg = classcol.reportinfo()
+        classcol = pytester.collect_by_name(modcol, "TestClass")
+        fspath, lineno, msg = classcol.reportinfo()  # type: ignore[union-attr]
```

Comment:
Can we add some `assert` here instead of the type-ignore?

Reply:
added assertions